testplan/testing/base.py

```diff
@@ -412,24 +511,115 @@ def start_test_resources(self):
+        use of assertions if the function accepts both ``env`` and ``result``
+        arguments.
+
+        These functions are also run within report error logging context,
+        meaning that if something goes wrong we will have the stack trace
+        in the final report.
         """
+
+        if not hook:
+            return
+
+        suite_report = TestGroupReport(
+            name=label,
+            category=ReportCategories.TESTSUITE,
+            suite_related=True,
+            # TODO: use synthesized instead of suite_related
+            # category=ReportCategories.SYNTHESIZED,
+        )
+        if self.result.report.has_uid(label):
+            self.result.report[label] = suite_report
+        else:
+            self.result.report.append(suite_report)
+
+        case_report = TestCaseReport(
+            name=hook.__name__,
+            description=strings.get_docstring(hook),
+            suite_related=True,
+            # TODO: use synthesized instead of suite_related
+            # category=ReportCategories.SYNTHESIZED,
+        )
+        suite_report.append(case_report)
+        case_result = self.cfg.result(
+            stdout_style=self.stdout_style, _scratch=self.scratch
+        )
+        # TODO: we are not using runtime_environment here as it is multitest-only
+        try:
+            interface.check_signature(hook, ["env", "result"])
+            hook_args = (self.resources, case_result)
+        except interface.MethodSignatureMismatch:
+            interface.check_signature(hook, ["env"])
+            hook_args = (self.resources,)
+
+        with compose_contexts(
+            case_report.timer.record("run"),
+            case_report.logged_exceptions(),
+            self.watcher.save_covered_lines_to(self.report),
+        ):
+            hook(*hook_args)
+
+        case_report.extend(case_result.serialized_entries)
+        case_report.attachments.extend(case_result.attachments)
+
+        if self.get_stdout_style(case_report.passed).display_testcase:
+            self.log_testcase_status(case_report)
+
+        case_report.pass_if_empty()
+        case_report.runtime_status = RuntimeStatus.FINISHED
+        suite_report.runtime_status = RuntimeStatus.FINISHED
+
+    def _dry_run_resource_hook(self, hook, label):
+
+        if not hook:
+            return
+
+        suite_report = TestGroupReport(
+            name=label,
+            category=ReportCategories.TESTSUITE,
+            suite_related=True,
+            # TODO: use synthesized instead of suite_related
+            # category=ReportCategories.SYNTHESIZED,
+        )
+        case_report = TestCaseReport(
+            name=hook.__name__,
+            suite_related=True,
+            # TODO: use synthesized instead of suite_related
+            # category=ReportCategories.SYNTHESIZED,
+        )
+        suite_report.append(case_report)
+        self.result.report.append(suite_report)
```

Comment:
Is not the duplicated label case ned to be handled here, or thta is just needed in the real run?

If the duplicate label case is needed here as well, can this be refactored to share this logic?

Reply:
the dry_run always start with a         

self.result.report = self._new_test_report()

thus we won't have dup label issue - which only happens in interactive mode.